apis/python/src/tiledb/vector_search/object_readers/directory_reader.py

```diff
@@ -67,25 +287,34 @@ def partition_class_name(self) -> str:
     def metadata_array_uri(self) -> str:
         return None
 
-    def list_paths(self, vfs: tiledb.VFS, uri):
-        children = vfs.ls(uri)
-        uri_path = Path(uri)
-        for child in children:
-            child_path = Path(child)
-            if child_path == uri_path:
-                continue
-            if self.exclude:
-                if any(child_path.match(exclude_glob) for exclude_glob in self.exclude):
-                    continue
-            if vfs.is_file(child):
-                if not child_path.match(self.glob):
-                    continue
-
-                if self.suffixes and child_path.suffix not in self.suffixes:
-                    continue
-                self.paths.append(child)
-            else:
-                self.list_paths(vfs, child)
+    def list_paths(self):
+        if self.search_uri.startswith("s3://"):
```

Comment:
We should use VFS for everything and not use boto3.

Reply:
Commented on this above.